agent/consul/server.go

```diff
@@ -467,10 +471,19 @@ func NewServer(config *Config, flat Deps, externalGRPCServer *grpc.Server) (*Ser
 	})
 
 	// TODO(NET-1380, NET-1381): thread this into the net/rpc and gRPC interceptors.
-	s.incomingRPCLimiter = rpcRate.NewHandler(rpcRate.HandlerConfig{
-		// TODO(NET-1379): pass in _real_ configuration.
-		GlobalMode: rpcRate.ModePermissive,
-	}, s)
+	if s.incomingRPCLimiter == nil {
+		s.incomingRPCLimiter = rpcRate.NewHandler(rpcRate.HandlerConfig{
+			GlobalMode: rpcRate.RequestLimitsModeFromNameWithDefault(config.RequestLimitsMode),
+			GlobalReadConfig: multilimiter.LimiterConfig{
+				Rate:  config.RequestLimitsReadRate,
+				Burst: int(config.RequestLimitsReadRate) * requestLimitsBurstMultiplier,
+			},
+			GlobalWriteConfig: multilimiter.LimiterConfig{
+				Rate:  config.RequestLimitsWriteRate,
+				Burst: int(config.RequestLimitsWriteRate) * requestLimitsBurstMultiplier,
+			},
+		}, s)
```

Comment:
Rather than duplicating this in the reload path, could we extract a conversion function?

Reply:
this is fixed.  🙏